### DISCUSSION TITLE
Some community feedback I've gathered about the current state of collections in lazer.

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#17168] [Ideas]
I made a request in a discord for collection manager to support lazer and overnight someone made a big comment about the shortcomings of the current implementation of collections. While it did seem somewhat venty, some very good points were made and I feel they deserved to be brought up for further discussion.
I'll be quoting or copy/pasting from here:

"Heres a few reasons it sucks currently:
1. Theres no collection "container" view so you cant just browse through your collections
2. Adding or removing maps from a collection requires right clicking and using the tiny dropdown menu which is stupid. It forces people who navigate ingame with tablet to use a mouse and it is just so tiny in general that it cant be done really fast without thinking. 
3. The F3 collections view doesnt allow adding or removing maps from collections (would mostly fix subject 2)
4. To open a collection you must use a small dropdown which is awful. If you have tons of collections the dropdown can even go off screen and require you to scroll it which can make it extremely easy to click wrong if you use a tablet to navigate. And due to not using a collection view with containers you gotta go right back to the dropdown if you click wrong.
5. This is just kinda stupid and im sure it will be fixed eventually but lazer doesnt accept stable collections files unless you rewrite the version number in the file."

On the matter of subject 1 I highly agree there should be at least an option for container view on collections.
For subject 2 I like the dropdown menu tho I think the scale needs to be slightly larger to support bigger buttons for tablet users and also I think having another button for adding the whole set should be an option like there is in stable especially considering sort by difficulty will be implemented sometime soon. 
Subject 3 I agree that this menu needs to be more feature rich given the screen real estate it takes up when open compared to all other collection related menus.
In terms of subject 4 I have heard many similar complaints from friends of mine and in multiplayer chats.
The fifth opinion is more in reference to collection manager functionality so this can likely be ignored.

Let me know your thoughts on the matter I'll be sure to update this discussion if any other noteworthy complaints come to my attention.

**Comment by bdach** [comment]
collections UI was mostly an initial makeshift effort to get *something* in place

ad. 1: #4919
ad. 5: i have no idea what "accept stable collections files" means or why messing with "version number" would do anything

**Reply by Kuuuube** [reply]
Worded it a bit poorly... 

Currently lazer uses the same collection format that stable does. However, it rejects stable collection files (worth noting that stable accepts lazer collection files) due to header issues and when changing that the collections work. It seems unintended but it makes using 3rd party tools on lazer collections harder.

**Reply by Kuuuube** [reply]
And by accepts and rejects I mean when replacing the `collection.db` and whether the game recognizes it or not after.

**Reply by Kuuuube** [reply]
It has been awhile since I last properly tested this and from further investigations it seems that was fixed already and I didn't realize lazer "prunes" maps out of collections that are not loaded into lazer and it found nothing in all my collections since I keep very few maps in lazer.

Suggestion 5 is not an issue anymore. Sorry about the confusion there.

**Reply by peppy** [reply]
you're using external tools which break the file and you expect it to work? that's not how software works.

please have those tool fix their issues. or at very least open a discussion specifically about that issue.

**Reply by Kuuuube** [reply]
As I said, I did not test this properly before mentioning it and this "bug" was already fixed in lazer. In the past there were some issues with manually making changes to collections in lazer. Those issues no longer exist.

I did not expect my discord ramblings to be copy-pasted into a discussion and from a quick test something seemed to be bugged. That turned out to be me misunderstanding that lazer removes maps from collections if they are not loaded into the game and has nothing to do with the initial issue I was trying to mention nor is it a bug.

The 5th point I made here is totally irrelevant now. I hope the rest can get some consideration though.

**Comment by Non-Binary-Programmer** [comment]
Another suggestion I have is to make the dropdown menu for adding beatmaps to a collection after right clicking the beatmap respect the user-defined collection order. Currently it just sorts by order created, which is very unhelpful.
